src/main/java/Dan/Parser.java

```diff
@@ -0,0 +1,57 @@
+package dan;
+
+import java.time.format.DateTimeParseException;
+
+public class Parser {
+    private TaskList tasks;
+
+    public Parser(TaskList tasks) {
+        this.tasks = tasks;
+    }
+    
+    public boolean parse(String input) {
+        String action = input.split(" ")[0];
+        try {
+            switch (action) {
+            case "bye":
+                Ui.sayonara();
+                return true;
+
+            case "list":
+                tasks.showTasks();
+                break;
+
+            case "mark":
+                tasks.markTask(Integer.parseInt(input.split(" ")[1]));
+                break;
+
+            case "unmark":
+                tasks.unMarkTask(Integer.parseInt(input.split(" ")[1]));
```

Comment:
Since method names should be in camelCase, perhaps this should be `tasks.unmarkTask()` instead? Because I see unmark as one word.

Reply:
You are right. Thought that unMark were 2 different words